<?xml version="1.0"?>
 <launch>
 	<arg name="sim" default="false"/>
    <arg name="local" default="true"/>

  <machine name="movo1" address="movo1" user="movo" 
               env-loader="/home/movo/env.sh" default="true" unless="$(arg local)"/>
 	
	<!-- Navigation configuration -->
	<include file="$(find movo_demos)/launch/nav/sensor_nav.launch">
		<arg name="local" value="$(arg local)"/>
		<arg name="sim" value="$(arg sim)"/>
	</include>
	
	<!-- Start Perception -->
    <node name="basic_grasping_perception" pkg="simple_grasping" type="basic_grasping_perception" output="screen">
    	<remap from="/head_camera/depth_registered/points" to="/movo_camera/point_cloud/points"/>
    	<!--remap unless="$(arg sim)" from="/head_camera/depth_registered/points" to="/movo_camera/points"/-->
    	<param name="simulated_kinova" value="$(arg sim)"/>
    	<param name="use_debug" value="true"/>
        <rosparam command="load" file="$(find movo_demos)/config/simple_grasping.yaml"/>
        
        <rosparam if="$(optenv MOVO_HAS_KINOVA_ARM_6DOF false)" command="load" file="$(find movo_moveit_config)/config/kinematics.yaml"/>
        <rosparam if="$(optenv MOVO_HAS_KINOVA_ARM_7DOF false)" command="load" file="$(find movo_7dof_moveit_config)/config/kinematics.yaml"/>
    </node>

    <!-- <node if="$(optenv MOVO_HAS_KINOVA_ARM_6DOF false)" name="inplace_pick_place_demo" pkg="movo_demos" type="inplace_pick_place_demo.py" output="screen">
    	<rosparam command="load" file="$(find movo_moveit_config)/config/kinematics.yaml"/>
    	<param name="table_height" value="0.74"/>
		<param name="sim" value="$(arg sim)"/>
		<param name="jaco_dof" value="6dof"/>
	</node>
	
	<node if="$(optenv MOVO_HAS_KINOVA_ARM_7DOF false)" name="inplace_pick_place_demo" pkg="movo_demos" type="inplace_pick_place_demo.py" output="screen">
    	<rosparam command="load" file="$(find movo_7dof_moveit_config)/config/kinematics.yaml"/>
    	<param name="table_height" value="0.74"/>
		<param name="sim" value="$(arg sim)"/>
		<param name="jaco_dof" value="7dof"/>
	</node> -->

	<node if="$(optenv MOVO_HAS_KINOVA_ARM_6DOF false)" name="inplace_pick_place_demo" pkg="movo_demos" type="pick_place_evolved.py" output="screen">
    	<rosparam command="load" file="$(find movo_moveit_config)/config/kinematics.yaml"/>
    	<param name="table_height" value="0.74"/>
		<param name="number_of_things_to_grab" value="2"/>
		<param name="sim" value="$(arg sim)"/>
		<param name="jaco_dof" value="6dof"/>
	</node>
	
	<node if="$(optenv MOVO_HAS_KINOVA_ARM_7DOF false)" name="inplace_pick_place_demo" pkg="movo_demos" type="pick_place_evolved.py" output="screen">
    	<rosparam command="load" file="$(find movo_7dof_moveit_config)/config/kinematics.yaml"/>
    	<param name="table_height" value="0.74"/>
		<param name="number_of_things_to_grab" value="2"/>
		<param name="sim" value="$(arg sim)"/>
		<param name="jaco_dof" value="7dof"/>
	</node>
    
</launch>
